### PR TITLE
Add CI and update linker flags

### PR DESCRIPTION
**Pull request by petricm**
Added the classic CI and add diagnostic color if compiler supports and proper treatment of `-Wl,--no-undefined`

### CODE REVIEW

File: cmake/ilcsoft_default_cxx_flags.cmake [hunk 1]
```diff
@@ -35,3 +39,9 @@ IF( USE_CXX11 )
 ELSE()
   MESSAGE( STATUS "NOT building with CXX11 standard" )
 ENDIF()
+
+
+CHECK_CXX_COMPILER_FLAG("-fdiagnostics-color=always" CXX_FLAG_WORKS_-fdiagnostics-color)
```
Comment (andresailer) [inline cmake/ilcsoft_default_cxx_flags.cmake]: `color=auto` please

**Comment by petricm** [comment]
Done

**Comment by petricm** [comment]
But this is not good now, since you don't see it in ninja

**Comment by andresailer** [comment]
Then export it explicitly in the CI cmake

**Comment by petricm** [comment]
ok, by default now it's auto, and in the CI build it's forced to always

**Pull request merged**